### PR TITLE
Add interface for specifying brancher labels and settings

### PR DESCRIPTION
**Pull request by tdgroot**
The configuration can be done as follows:

```php
$testStage = $configuration->addStage('test', 'example.com');
$testStage->addBrancherServer('appname')
    ->setLabels(['stage=test', 'ci_ref=' . \getenv('GITHUB_REF') ?: 'none'])
    ->setSettings(['cron_enabled' => false, 'supervisor_enabled' => false]);
```

These labels can be used in Hypernode Deploy to determine what brancher nodes to clean up, for example: clean up brancher servers matching specified stage and their labels.

### CODE REVIEW

**Review by martijneichelsheim — APPROVED** [review]

File: templates/deploy.magento2.php [hunk 1]
```diff
@@ -14,6 +14,8 @@
 $productionStage->addServer('appname.hypernode.io');
 
 $testStage = $configuration->addStage('test', 'example.com');
-$testStage->addBrancherServer('appname');
+$testStage->addBrancherServer('appname')
+    ->setLabels(['stage=test', 'ci_ref=' . \getenv('GITHUB_RUN_ID') ?: 'none'])
```
Comment (AlexanderGrooff) [inline templates/deploy.magento2.php:18]: These labels are all formatted as `X=Y`, but there isn't some validation based on this format right? Any format of label is allowed?

File: src/BrancherServer.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        return $this->getOptions()[self::OPTION_HN_BRANCHER_LABELS] ?? [];
+    }
+
+    /**
+     * @param string[] $labels Labels to be applied to the brancher node
+     * @return $this
+     */
+    public function setLabels(array $labels): self
+    {
+        $this->setOption(self::OPTION_HN_BRANCHER_LABELS, $labels);
```
Comment (AlexanderGrooff) [inline src/BrancherServer.php:20]: Do you think it's wise to add input validation in the client here or should we let the API validate the data?

**Review by AlexanderGrooff — APPROVED** [review]
Looks good!

File: templates/deploy.magento2.php (hunk 1 above)
Comment (tdgroot) [reply templates/deploy.magento2.php:18]: Any format is allowed indeed

File: src/BrancherServer.php (hunk 2 above)
Comment (tdgroot) [reply src/BrancherServer.php:20]: Yes I think so, since this is a very light component, let's add the validation in hypernode-deploy

**Pull request merged**